.travis.yml

```diff
@@ -1,52 +1,58 @@
 dist: xenial
 sudo: false
 language: java
-matrix:
+if: tag is blank
+jobs:
   include:
-  - jdk: openjdk11
-    env: TRIPLEA_RELEASE=true
-  - jdk: openjdk12
-    env: TRIPLEA_RELEASE=false
-addons:
-  postgresql: "10"
-  apt:
-    packages:
-      # python for map file splitter
-      - python3
-      - python3-yaml
-install:
-- echo "================ Build step 'install' =================" > /dev/null
-## shellcheck all shell scripts
--  find .travis/ -type f | xargs grep -lE "^#\!/bin/bash$" | xargs shellcheck
-## validate ASAP so syntax, checkstyle or unit tests fail the build early
-- ./gradlew validateYamls checkstyleMain checkstyleTest spotlessCheck test
-- ./.travis/install
-before_script:
-- echo "================ Build step 'before_script' =================" > /dev/null
-- ./.travis/setup_gpg
-script:
-- echo "================ Build step 'script' =================" > /dev/null
-- ./gradlew check jacocoTestReport
-after_success:
-- echo "================ Build step 'after_success'' =================" > /dev/null
-- bash <(curl -s https://codecov.io/bash)  # upload coverage report - https://github.com/codecov/example-gradle
+    - stage: verify
+      env: DESCRIPTION="static verification"
+      jdk: openjdk11
+      script:
+        ## shellcheck all shell scripts
+        - find .travis/ -type f | xargs grep -lE "^#\!/bin/bash$" | xargs shellcheck
+        - ./gradlew --parallel validateYamls spotlessCheck checkstyleMain checkstyleTest pmdMain
+    - stage: verify
+      env: DESCRIPTION=integTest
+      jdk: openjdk11
+      addons: {postgresql: "10"}
+      script:
+        - ./.travis/setup_database &
+        - ./.travis/launch_http_server
+        - ./gradlew integTest
+    - stage: verify
+      env: DESCRIPTION="JDK12 unitTests"
+      jdk: openjdk12
+      script:
+        - ./gradlew --parallel test
```

Comment:
I mean obviously we don't need to run checkstyle here again, but I think integration tests, might be an equally good indicator that something breaks in a future version

Reply:
IIRC error prone gave us the most trouble on upgrade, so we're covered there running that.

integTests I'm really on the fence. If we can make it a bit faster than the JDK11 integ test by cutting a step, that would be excellent. Integ tests being the slowest build is what we wait for, if the JDK12 VM keeps taking a bit longer to spin up, it could push out the total build time, hence I'm on the fence about it.

Reply:
I was thinking to go ahead and see how this works and we can update it. Early results are bad.

1) integ tests are not reliable in parallel right now: https://github.com/triplea-game/triplea/issues/5163

If we run both tests and integTests, then the tests will be run in non-parallel mode and would be pretty slow. We could execute the test and 'integTest' as two different commands, that would make things a bit slower again.

2) Results for runtime are not good. integTest is a slow task, for JDK11 builds the test and integTest were broken up to help speed up the overall build time. Having those two tasks together adds to our slowest task.

In terms of performance results, this build gave it a try: https://travis-ci.org/DanVanAtta/triplea/builds/579469507; The JDK12 test + integTest task took 9m43s. The JDK11 integTest task took 8m5s. That represents a 20% increase in test speed, we'd expect perhaps even slower if we turned off the parallel flag.